File: Problems & Solutions/1470. Shuffle the Array/06 February 2023/Approach 2 - In-place Filling/Shuffle the Array.cpp
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        for (int i = n; i < 2 * n; ++i) {
            int secondNum = nums[i] << 10;
            nums[i - n] |= secondNum;
        }

        int allOnes = pow(2, 10) - 1;

        for (int i = n - 1; i >= 0; --i) {
            int secondNum = nums[i] >> 10;
            int firstNum = nums[i] & allOnes;
            nums[2 * i + 1] = secondNum;
            nums[2 * i] = firstNum;
        }
        return nums;
    }
};

File: Problems & Solutions/1470. Shuffle the Array/06 February 2023/Approach 2 - In-place Filling/Shuffle the Array.py
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        for i in range(n, 2 * n):
            secondNum = nums[i] << 10
            nums[i - n] |= secondNum

        allOnes = int(pow(2, 10)) - 1

        for i in range(n - 1, -1, -1):
            secondNum = nums[i] >> 10
            firstNum = nums[i] & allOnes
            nums[2 * i + 1] = secondNum
            nums[2 * i] = firstNum
        return nums